### DISCUSSION TITLE
unit testing code inside prisma transaction

### DISCUSSION THREAD

**Discussion by msreekm** [prisma/prisma#14050] [Q&A] [answered]
await this.prisma?.$transaction(async prisma => {
     await prisma.user.create();
    await prisma.userRole.create();
})

how do I test this transaction ?
below code does not work. any suggestions?
expect(prismaMock.user.create).toHaveBeenCalledWith({})

**Comment by austincrim** [comment, marked as answer]
Hey @msreekm 👋,

This is something I haven't tried yet, but have you attempted to mock the `$transaction` method?

**Reply by msreekm** [reply]
@austincrim  , thanks for response, any sample code I can refer ?

**Reply by pantharshit00** [reply]
Hello @msreekm,

Sorry for late reply. I have answered a similar question recently. You can refer here on how you can mock interactive transactions: https://github.com/prisma/prisma/discussions/14435#discussioncomment-3857014

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
